lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php

```diff
@@ -639,6 +639,15 @@ class ClassMetadataInfo implements ClassMetadata
      */
     public $containsForeignIdentifier = false;
 
+    /**
+     * READ-ONLY: Flag indicating whether the identifier/primary key contains at least one ENUM type.
+     *
+     * This flag is necessary because some code blocks require special treatment of this cases.
+     *
+     * @var bool
+     */
+    public $containsEnumIdentifier = false;
```

Comment:
Every new cm field needs to be handled in seriamization sleep code otherwise it gets lost in cache

Reply:
Commit [98ecc21](https://github.com/doctrine/orm/pull/9629/commits/98ecc2111b00f0e5e06212d7160a5ee5563efdea)